include/arch/arm/arch/64/mode/object/structures.bf

```diff
@@ -65,8 +65,8 @@ block page_upper_directory_cap {
     field capPUDIsMapped             1
     field_high capPUDMappedAddress   10
 #if defined (CONFIG_ARM_SMMU)  && defined (AARCH64_VSPACE_S2_START_L1)
-    field capPUDMappedCB             12
-    padding                          36
+    field capPUDMappedCB             8
+    padding                          40
```

Comment:
This commit also changes the width of the CB field in the caps. Was that intentional? If yes, should probably be mentioned in the commit message. I haven't checked all other `cb` occurrences if the new width fits or may lead to overflow.

Reply:
Intentional. According to the other PR and the SMMUv2 spec, there is a maximum number of context banks of 128 which requires 8 bits to include a valid bit. I tried to update the width in all of the other locations.  The asid_map_t is short of available bit space and I saw that there's already a push for changing them to 8bits in other PRs so figured to make this part of the change at the same time.

Reply:
Ok, that makes sense. Would be good to mention in the commit message. I have checked the other occurrences, and at least the types are safe everywhere, so this is good from my side.

Reply:
CONFIG_ARM_SMMU seems implying v2. We probably will need to support V3 soon.